### PR TITLE
Migration of "Catalog > Brand & Suppliers > Suppliers" listing

### PR DESCRIPTION
**Pull request by tomas862**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | supplier list can be reached via **admin-dev/index.php/sell/catalog/suppliers/**
| Type?         | new feature
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | https://github.com/PrestaShop/PrestaShop/issues/10545
| How to test?  | The list should work exactly the same just that the "Add", "View" and "Edit" links will point to legacy suppliers form page. Needs assets to be rebuilt.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/11153)

### CODE REVIEW

**Comment by tomas862** [comment]
@matks  ready for review :slightly_smiling_face:

File: classes/helper/HelperList.php
Comment (matks) [inline classes/helper/HelperList.php:322]: 👍

File: src/Adapter/Supplier/CommandHandler/AbstractDeleteSupplierHandler.php
Comment (matks) [inline src/Adapter/Supplier/CommandHandler/AbstractDeleteSupplierHandler.php]: 🤔 mmm

The idea seems nice however I have not (so far) seen the use of MySQL transactions in PrestaShop so there might be an historical reason why we cannot use them. Moreover, it only works for InnoDB SQL storage engines, not MyISAM ; and the configuration of the database is not in the scope of PrestaShop.

I need your insights @mickaelandrieu @PierreRambaud @Quetzacoalt91 @eternoendless

File: src/Core/Grid/Definition/Factory/SupplierGridDefinitionFactory.php
Comment (matks) [inline src/Core/Grid/Definition/Factory/SupplierGridDefinitionFactory.php]: You should be able to pass the route name now, as in https://github.com/PrestaShop/PrestaShop/blob/5b2f730c60a611455b6b14d411f379c6eed7362b/src/Core/Grid/Definition/Factory/CustomerGridDefinitionFactory.php#L301

File: src/PrestaShopBundle/Controller/Admin/Sell/Catalog/SupplierController.php
Comment (matks) [inline src/PrestaShopBundle/Controller/Admin/Sell/Catalog/SupplierController.php]: Please use `$this->render()`

File: src/PrestaShopBundle/Resources/config/routing/admin/sell/catalog/suppliers.yml
Comment (matks) [inline src/PrestaShopBundle/Resources/config/routing/admin/sell/catalog/suppliers.yml]: Please use path `/new`

File: src/PrestaShopBundle/Resources/config/services/adapter/supplier.yml
Comment (matks) [inline src/PrestaShopBundle/Resources/config/services/adapter/supplier.yml:20]: I think you do not need to declare this service (and it cannot be used as it is abstract)

**Review by matks — CHANGES_REQUESTED** [review]
Hello @tomas862 nice work ! A few minor feedbacks here 😃

File: src/Adapter/Supplier/CommandHandler/AbstractDeleteSupplierHandler.php
Comment (PierreRambaud) [reply src/Adapter/Supplier/CommandHandler/AbstractDeleteSupplierHandler.php]: Bad idea,
1) We need to support MyISAM
2) This is an external object which will play with transaction. (Like the Db itself)
3) Let's go move to PostgreSQL :trollface: 
4) Ok => :door:

Comment (matks) [reply src/Adapter/Supplier/CommandHandler/AbstractDeleteSupplierHandler.php]: I would love to support PostgreSQL for PrestaShop ❤️

File: src/PrestaShopBundle/Resources/config/services/adapter/supplier.yml
Comment (tomas862) [reply src/PrestaShopBundle/Resources/config/services/adapter/supplier.yml:20]: I'm using it in two command handlers like this defining this service as a parent:

``` yml
prestashop.adapter.supplier.command_handler.delete_supplier_handler:
    class: PrestaShop\PrestaShop\Adapter\Supplier\CommandHandler\DeleteSupplierHandler
    parent: prestashop.adapter.supplier.command_handler.abstract_delete_supplier_handler
    tags:
      - name: tactician.handler
        command: PrestaShop\PrestaShop\Core\Domain\Supplier\Command\DeleteSupplierCommand
    public: true
```

however, since this abstraction is not supposed to be used directly I'll add an argument:

``` yml
public: false
```

Thanks for notice :+1:

**Comment by tomas862** [comment]
added some changes regarding the feedback! :+1:

**Review by matks — DISMISSED** [review]

**Comment by matks** [comment]
@tomas862 great !
Let's get QA approval and only then rebase 😄

**Comment by marionf** [comment]
Hello @tomas862 

When I click on import I have this error:

capture d ecran_781

@TristanLDD is it ok for you ?

capture d ecran_783

**Comment by tomas862** [comment]
Hi @marionf

About the error - the problem happens on import page so its totally not related with current pull request. Someone will fix that error on separate pull request

**Comment by TristanLDD** [comment]
Hello, 
There are a few corrections to do on this page: 

- Add placeholders in search fields

- Align the search button and the "Actions" label on the right

- Vertically align the pagination system components 

capture d ecran 2018-12-20 a 10 32 55

Thanks !

**Comment by marionf** [comment]
Removing QA approved, I will retest it when changes requested by @TristanLDD will be applied

**Comment by tomas862** [comment]
Changes made:

* Vertically align the pagination system components:
pagination_text

* adds placeholders  (except for active selector @matks will inform me if we will do that)
placeholders

* for moving search button to the right - https://github.com/PrestaShop/PrestaShop/pull/11877 has to be merged first ;)

**Review by matks — APPROVED** [review]

**Comment by matks** [comment]
Thanks @tomas862

**Pull request merged**
